### PR TITLE
Resources/application properties

### PR DESCRIPTION
**Pull request by oscarlopez8**

### CODE REVIEW

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 1]
```diff
@@ -65,6 +68,8 @@
      * @param request contains client request data.
      * @return Response is the result of the conversion.
      */
+    @Autowired
+    FileRepository fileRepository;
     @PostMapping("/videoConverter")
     @ApiOperation(value = "Video specifications", notes = "Provides values for converting Video file to other one.",
             response = Response.class)
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: Review the limit lenght in code line: 81

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 2]
```diff
@@ -105,10 +119,11 @@ public Response videoConverter(
                 fileResponse = videoConverter.convert(videoParam);
                 LinkGenerator linkGenerator = new LinkGenerator();
                 fileResponse.setDownload(linkGenerator.linkGenerator(fileResponse.getDownload(), request));
-            }
-            else {
+
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: Delete this code line

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 3]
```diff
@@ -105,10 +119,11 @@ public Response videoConverter(
...
+            } else {
                 throw new Md5Exception(ErrorMessageJfc.MD5_ERROR.getErrorMessageJfc(), videoParam.getMd5());
             }
+
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: Delete this code line please

File: src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import javax.persistence.Id;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+@Entity
+
+public class FileEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    // Content id value.
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java:29]: Add a code line here

File: src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
+package com.jalasoft.jfc.model.entity;
```
Comment (Enrique-C) [inline src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java:9]: Add copy right

File: src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java [hunk 6]
```diff
@@ -58,6 +61,9 @@
      * @param request is client request value.
      * @return ResponseEntity<Response> for status code.
      */
+    // Inject FileRepository
+    @Autowired
+    FileRepository fileRepository;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java]: add new line leave this line of code.

File: src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java [hunk 7]
```diff
@@ -70,6 +76,17 @@
...
+            FileEntity fileEntity = new FileEntity();
+
+            if (fileRepository.findByMd5(Md5Checksum.getMd5(fileUploaded)) != null) {
+                param.setInputPathFile(fileRepository.findByMd5(Md5Checksum.getMd5(fileUploaded)).getFilePath());
+            } else {
+                param.setInputPathFile(fileUploaded);
+                fileEntity.setFilePath(fileUploaded);
+                fileEntity.setMd5(Md5Checksum.getMd5(fileUploaded));
+                fileRepository.save(fileEntity);
+            }
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java:93]: add a new line leave this line of code.

File: src/main/java/com/jalasoft/jfc/controller/AudioConverterController.java [hunk 8]
```diff
@@ -60,6 +63,10 @@
      * @param request contains client request data.
      * @return Response is the result of the conversion.
      */
+
+    // Inject FileRepository
+    @Autowired
+    FileRepository fileRepository;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/AudioConverterController.java]: add new line leave this one.

File: src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java [hunk 9]
```diff
@@ -65,6 +68,10 @@
      * @param request contains client request data.
      * @return Response is the result of the conversion.
      */
+
+    // Inject FileRepository
+    @Autowired
+    FileRepository fileRepository;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/ImageConverterController.java]: add new line leave this one.

File: src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java [hunk 10]
```diff
@@ -69,6 +72,9 @@
      * @param isMetadata boolean of metadata.
      * @return Response is the result of the conversion.
      */
+    // Inject FileRepository
+    @Autowired
+    FileRepository fileRepository;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/PdfConverterController.java]: add a new line leave this one.

File: src/main/java/com/jalasoft/jfc/controller/PptxConverterController.java [hunk 11]
```diff
@@ -64,6 +67,10 @@
      * @param isMetadata boolean of metadata.
      * @return Response is the result of the conversion.
      */
+
+    // Inject FileRepository
+    @Autowired
+    FileRepository fileRepository;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/PptxConverterController.java]: add a new line leve this one.

File: src/main/java/com/jalasoft/jfc/controller/PptxConverterController.java [hunk 12]
```diff
@@ -84,10 +91,19 @@ public Response pptxConverterToPdf(
...
+
             if (Md5Checksum.getMd5(fileUploadedPath, md5)) {
+                if (fileRepository.findByMd5(md5) != null) {
+                    pptxParam.setInputPathFile(fileRepository.findByMd5(md5).getFilePath());
+                } else {
+                    pptxParam.setInputPathFile(fileUploadedPath);
+                    fileEntity.setFilePath(fileUploadedPath);
+                    fileEntity.setMd5(md5);
+                    fileRepository.save(fileEntity);
+                }
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/PptxConverterController.java:101]: add a new line leave this line.

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 13]
```diff
@@ -65,6 +68,10 @@
      * @param request contains client request data.
      * @return Response is the result of the conversion.
      */
+
+    // Inject FileRepository
+    @Autowired
+    FileRepository fileRepository;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java]: add a new line leave this line of code.

File: src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java [hunk 14]
```diff
@@ -86,9 +93,18 @@ public Response videoConverter(
...
+
             if (Md5Checksum.getMd5(fileUploadedPath, md5)) {
+                if (fileRepository.findByMd5(md5) != null) {
+                    videoParam.setInputPathFile(fileRepository.findByMd5(md5).getFilePath());
+                } else {
+                    videoParam.setInputPathFile(fileUploadedPath);
+                    fileEntity.setFilePath(fileUploadedPath);
+                    fileEntity.setMd5(md5);
+                    fileRepository.save(fileEntity);
+                }
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/VideoConverterController.java:107]: add a new line leave this one.

File: src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java [hunk 15]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2019 Jalasoft.
+ *
+ * This software is the confidential and proprietary information of Jalasoft.
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java:8]: add a new line leave this line of code.

File: src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+
+/**
+ * Defines UserRepository.
+ *
+ * @version 0.1 Jan 2020.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java]: add number of the day.

File: src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * @author Oscar Lopez.
+ */
+@Entity
+public class FileEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java:27]: add a new line leave this one.

File: src/main/java/com/jalasoft/jfc/model/repository/FileRepository.java [hunk 18]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2019 Jalasoft.
+ *
+ * This software is the confidential and proprietary information of Jalasoft.
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/repository/FileRepository.java:8]: add new line leave this line of code.

File: src/main/java/com/jalasoft/jfc/model/repository/FileRepository.java [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+package com.jalasoft.jfc.model.repository;
+
+import com.jalasoft.jfc.model.entity.FileEntity;
+import org.springframework.data.jpa.repository.Query;
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/repository/FileRepository.java]: separate this lines of code.

File: src/main/java/com/jalasoft/jfc/model/repository/FileRepository.java [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+/**
+ * Defines FileRepository.
+ *
+ * @version 0.1 Jan 2020.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/repository/FileRepository.java]: add number of the day.

File: src/main/java/com/jalasoft/jfc/model/entity/UserEntity.java [hunk 21]
```diff
@@ -19,7 +19,7 @@
 /**
  * Defines UserEntity.
  *
- * @version 0.1 Jan 2020.
+ * @version 0.1 10 Jan 2020.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/entity/UserEntity.java:22]: today is 13

File: src/main/java/com/jalasoft/jfc/model/repository/UserRepository.java [hunk 22]
```diff
@@ -17,7 +17,7 @@
 /**
  * Defines UserRepository.
  *
- * @version 0.1 Jan 2020.
+ * @version 0.1 10 Jan 2020.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/repository/UserRepository.java:20]: today is 13

**Review by JuanCarlosMartinez4 — COMMENTED** [review]
check please all comments

File: src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java [hunk 23]
```diff
@@ -0,0 +1,85 @@
...
+ */
+package com.jalasoft.jfc.model.entity;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+
+/**
+ * Defines UserRepository.
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/model/entity/FileEntity.java]: update this comment

File: src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java [hunk 24]
```diff
@@ -10,10 +10,14 @@
 package com.jalasoft.jfc.controller;
 
 import com.jalasoft.jfc.model.Param;
+import com.jalasoft.jfc.model.entity.FileEntity;
 import com.jalasoft.jfc.model.exception.ConvertException;
 import com.jalasoft.jfc.model.exception.Md5Exception;
 import com.jalasoft.jfc.model.exception.ZipJfcException;
 import com.jalasoft.jfc.model.metadata.MetadataConverter;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java]: delete this line.

File: src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java [hunk 25]
```diff
@@ -10,10 +10,14 @@
...
+import com.jalasoft.jfc.model.repository.FileRepository;
+
```
Comment (JuanCarlosMartinez4) [inline src/main/java/com/jalasoft/jfc/controller/MetadataConverterController.java]: delete this line

**Review by JuanCarlosMartinez4 — APPROVED** [review]

**Pull request merged**